test/integration/targets/group/tasks/tests.yml

```diff
@@ -0,0 +1,184 @@
+---
+##
+## group add
+##
+
+- name: create group (check mode)
+  group:
+    name: ansibullgroup
+    state: present
+  register: create_group_check
+  check_mode: True
+
+- name: get result of create group (check mode)
+  script: grouplist.sh "{{ ansible_distribution }}"
+  register: create_group_actual_check
+
+- name: assert create group (check mode)
+  assert:
+    that:
+    - create_group_check is changed
+    - '"ansibullgroup" not in create_group_actual_check.stdout_lines'
+
+- name: create group
+  group:
+    name: ansibullgroup
+    state: present
+  register: create_group
+
+- name: get result of create group
+  script: grouplist.sh "{{ ansible_distribution }}"
+  register: create_group_actual
+
+- name: assert create group
+  assert:
+    that:
+    - create_group is changed
+    - create_group.gid is defined
+    - '"ansibullgroup" in create_group_actual.stdout_lines'
+
+- name: create group (idempotent)
+  group:
+    name: ansibullgroup
+    state: present
+  register: create_group_again
+
+- name: assert create group (idempotent)
+  assert:
+    that:
+    - not create_group_again is changed
+
+##
+## group check
+##
+
+- name: run existing group check tests
+  group:
+    name: "{{ create_group_actual.stdout_lines|random }}"
+    state: present
+  with_sequence: start=1 end=5
+  register: group_test1
+
+- name: validate results for testcase 1
+  assert:
+      that:
+          - 'group_test1.results is defined'
+          - 'group_test1.results|length == 5'
+
+- name: validate change results for testcase 1 (jinja2 >= 2.6)
+  assert:
+      that:
+          - "group_test1.results|map(attribute='changed')|unique|list == [False]"
+          - "group_test1.results|map(attribute='state')|unique|list == ['present']"
+  when: "jinja2_version.stdout is version('2.6', '>=')"
+
+- name: validate change results for testcase 1 (jinja2 < 2.6)
+  assert:
+      that:
+          - "not group_test1.results[0]['changed']"
+          - "not group_test1.results[1]['changed']"
+          - "not group_test1.results[2]['changed']"
+          - "not group_test1.results[3]['changed']"
+          - "not group_test1.results[4]['changed']"
+  when: "jinja2_version.stdout is version('2.6', '<')"
+
+##
+## group add with gid
+##
+
+- name: create a group with a gid (check mode)
+  group:
+    name: ansibullgroup2
+    gid: 666
```

Comment:
Find an available gid instead of using a fixed value to avoid cases where it already exists.